src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -0,0 +1,814 @@
+				xocl_dbg(&sdm->pdev->dev, "Lower Critical Limit: 0x%x\r\n", *(srec->lower_critical_limit));
+				xocl_dbg(&sdm->pdev->dev, "Lower Critical Adjusted Limit : %f\r\n", (*(srec->lower_critical_limit)) *
+					   pow(10, srec->unit_modifier_byte));
+			}
+			if(srec->threshold_support_byte & THRESHOLD_LOWER_FATAL_MASK) {
+				xocl_dbg(&sdm->pdev->dev, "Lower Fatal Limit: 0x%x\r\n", *(srec->lower_fatal_limit));
+				xocl_dbg(&sdm->pdev->dev, "Lower Fatal Adjusted Limit : %f\r\n", (*(srec->lower_fatal_limit)) *
+					   pow(10, srec->unit_modifier_byte));
+			}
+		}
+		if(srec->base_unit_type_length != SDR_NULL_BYTE) {
+			uint8_t bu_type   = (srec->base_unit_type_length >> SDR_TYPE_POS) & SDR_TYPE_MASK;
+			if(bu_type == TYPECODE_ASCII)
+				xocl_dbg(&sdm->pdev->dev, "Base Unit Value : %s\r\n", srec->base_unit);
+			else if (bu_type == TYPECODE_BINARY)
+				xocl_dbg(&sdm->pdev->dev, "Sensor Base Unit : 0x%x\r\n", *(srec->base_unit));
+		}
+
+		uint8_t val_len = srec->value_type_length & SDR_LENGTH_MASK;
+		if(srec->unit_modifier_byte != 0x7F)
+		{
+			if(val_len == 1)
+				xocl_dbg(&sdm->pdev->dev, "Sensor Adjusted Value : %f\r\n", (*(uint8_t *)srec->value) *
+					   pow(10, srec->unit_modifier_byte));
+			else if(val_len == 2)
+				xocl_dbg(&sdm->pdev->dev, "Sensor Adjusted Value : %f\r\n", (*(uint16_t *)srec->value) *
+					   pow(10, srec->unit_modifier_byte));
+			else if(val_len == 4)
+				xocl_dbg(&sdm->pdev->dev, "Sensor Adjusted Value : %f\r\n", (*(uint32_t *)srec->value) *
+					   pow(10, srec->unit_modifier_byte));
+		}
+
+		if(srec->status == SENSOR_NOT_PRESENT)
+			xocl_dbg(&sdm->pdev->dev, "Sensor Status : Sensor Not Present\r\n");
+		else if(srec->status == SENSOR_PRESENT_AND_VALID)
+			xocl_dbg(&sdm->pdev->dev, "Sensor Status : Sensor Present and Valid\r\n");
+		else if(srec->status == DATA_NOT_AVAILABLE)
+			xocl_dbg(&sdm->pdev->dev, "Sensor Status : Sensor Data Not Available\r\n");
+		else if(srec->status == SENSOR_STATUS_NOT_AVAILABLE)
+			xocl_dbg(&sdm->pdev->dev, "Sensor Status : Sensor Data Not Available\r\n");
+
+		srec++;
+		sr--;
+
+		xocl_dbg(&sdm->pdev->dev, "\r\n");
+	}
+	xocl_dbg(&sdm->pdev->dev, "%s\r\n",  repo_object->eor.eor_marker);
+}
+
+static uint8_t sdr_get_id(uint8_t repo_type)
+{
+	int id = 0;
+
+	switch(repo_type) {
+		case SDR_TYPE_BDINFO: id = 0; break;
+		case SDR_TYPE_TEMP: id = 1; break;
+		case SDR_TYPE_VOLTAGE: id = 2; break;
+		case SDR_TYPE_CURRENT: id = 3; break;
+		case SDR_TYPE_POWER: id = 4; break;
+		case SDR_TYPE_QSFP: id = 5; break;
+		case SDR_TYPE_VPD_PCIE: id = 6; break;
+		case SDR_TYPE_IPMIFRU: id = 7; break;
+		case SDR_TYPE_CSDR_LOGDATA: id = 8; break;
+		case SDR_TYPE_VMC_LOGDATA: id = 9; break;
+		default: id = -1; break;
+	}
+	return id;
+}
+
+static struct sdr_response* parse_sdr_info(char *in_buf,
+                                           struct xocl_hwmon_sdm *sdm,
+                                           bool create_sysfs)
+{
+	int buf_index = SDR_NULL_BYTE;
```

Comment:
Is this from Sandeep's code? buf_index = 0,1,2?

Can we translate the index 0,1,2 to meaningful macro?
like
#define SDR_DEFAULT_IDX 0
etc...